src/app/pages/storage/modules/devices/stores/devices-store.service.ts

```diff
@@ -78,7 +78,7 @@ export class DevicesStore extends ComponentStore<DevicesState> {
         return this.ws.call('pool.query', [[['id', '=', poolId]]]).pipe(
           switchMap((pools) => {
             // TODO: Handle pool not found.
-            return this.ws.call('disk.query', [[['pool', '=', pools[0].name]], { extra: { pools: true } }]).pipe(
+            return this.ws.call('disk.query', [[['pool', '=', pools?.[0]?.name]], { extra: { pools: true } }]).pipe(
```

Comment:
If this data can be falsy, we should check beforehand. Otherwise, we will just make an invalid request to middleware which is guaranteed to fail.